src/privacy/privacy.ts

```diff
@@ -73,12 +53,15 @@ export const obfuscateData = (_data: any, fields: string[] | string) => {
   };
 };
 
-export const obfuscateDocument = (document: SignedDocument, fields: string[] | string): SignedDocument => {
+// TODO shouldn't document be any and shouldn't we validate first it's a valid document ?
```

Comment:
We should validate the schema first, but document shouldnt be any (so we can make use of it's type later), unless you want to cast it after it has been validated.

Reply:
it depends if we want to trust consumer or not
- if people use typescript then no problems
- if they dont they can pass any value, it may fail without explanation

We should at least leave document as SignedDocument for people using TS, but extra validation would be nice also IMHO)

Reply:
My concern is that whether people use TS or not, there might be type issue during runtime. Especially when we download and upload document all the time now.

Reply:
Yes that's what I mean by `extra validation would be nice`

=> https://github.com/Open-Attestation/open-attestation/issues/59

In the meantime, keep the type correct, better for TS :)